src/widget/wrecordingduration.cpp

```diff
@@ -0,0 +1,22 @@
+#include "widget/wrecordingduration.h"
+
+WRecordingDuration::WRecordingDuration(QWidget *parent,
+                                    RecordingManager* pRecordingManager)
+        : WLabel(parent),
+          m_durationRecordedStr(""),
+          m_pRecordingManager(pRecordingManager) {
+}
+
+WRecordingDuration::~WRecordingDuration() {
+}
+
+void WRecordingDuration::setup(const QDomNode& node, const SkinContext& context) {
+    WLabel::setup(node, context);
+    connect(m_pRecordingManager, SIGNAL(durationRecorded(QString)),
+        this, SLOT(refreshLabelText(QString)));
+}
+
+void WRecordingDuration::refreshLabelText(QString durationRecorded) {
+    m_durationRecordedStr = durationRecorded;
```

Comment:
Will change. As I learned `durationRecorded` can be passed to setText directly if it shouldn't be stored for any other purpose.
Also I would like to clear the label when recording is stopped, so it doesn't show last duration when started again.